### PR TITLE
VZ-7506: Make CLI resilient to streaming console output during install/upgrade

### PR DESCRIPTION
**Pull request by mgianatagh**
During install/upgrade if the connection is lost to streaming console output, the CLI use to just exit.  The install or upgrade continues, but environments like our pipelines think the install has failed because the CLI has exited with an error.

This PR has changes to have the CLI make up to 5 attempts to reconnect.  For example, if the VPO pod recycles during install, the CLI will automatically re-establish the connection to the new VPO pod and continue streaming the console output.

### CODE REVIEW

File: tools/vz/cmd/helpers/vpo.go [hunk 1]
```diff
@@ -207,18 +203,42 @@ func WaitForOperationToComplete(client clipkg.Client, kubeClient kubernetes.Inte
...
+					time.Sleep(5 * time.Second)
+					continue
+				}
+				connectRetryCount = 0
+				sc.Split(bufio.ScanLines)
+			}
+
+			scannedOk := sc.Scan()
+			if !scannedOk {
+				fmt.Fprintf(outputStream, "Lost connection to the console output, attempting to reconnect\n")
```
Comment (bobdonat) [inline tools/vz/cmd/helpers/vpo.go:230]: Should the error be shown here too? I think sc.Err() with return the error.

**Comment by bobdonat** [comment]
Is his same change needed for uninstall

File: tools/vz/cmd/helpers/vpo.go (hunk 1 above)
Comment (mgianatagh) [reply tools/vz/cmd/helpers/vpo.go:230]: sc.Err was always blank for me.  I guess I could check if it was non-nil and tailor the message based on that.

File: tools/vz/cmd/helpers/vpo.go [hunk 2]
```diff
@@ -207,18 +203,42 @@ func WaitForOperationToComplete(client clipkg.Client, kubeClient kubernetes.Inte
...
+		const maxConnectRetries = 5
+
 		for {
-			sc.Scan()
+			if sc == nil {
+				sc, err = getScanner(client, kubeClient)
+				if err != nil {
+					connectRetryCount++
+					fmt.Fprintf(outputStream, fmt.Sprintf("Failed to connect to the console output, retry %d of %d: %v\n", connectRetryCount, maxConnectRetries, err))
+					if connectRetryCount >= 5 {
```
Comment (michael-cico) [inline tools/vz/cmd/helpers/vpo.go:218]: the retry count/interval has a max of 25 secs cumulatively, which may not be enough if a pod's moving etc.

File: tools/vz/cmd/helpers/vpo.go (hunk 1 above)
Comment (michael-cico) [reply tools/vz/cmd/helpers/vpo.go:230]: I think the problem I saw during uninstall also was that the `Scan` call hung until the operation finished or some other timeout kicked in, and there was no error.  So this impl is able to pick back up when the pod is whacked?

File: tools/vz/cmd/install/install.go [hunk 3]
```diff
@@ -220,7 +214,7 @@ func runCmdInstall(cmd *cobra.Command, args []string, vzHelper helpers.VZHelper)
 	}
 
 	// Wait for the Verrazzano install to complete
-	return waitForInstallToComplete(client, kubeClient, vzHelper, vpoPodName, types.NamespacedName{Namespace: vzNamespace, Name: vzName}, timeout, logFormat)
+	return waitForInstallToComplete(client, kubeClient, vzHelper, types.NamespacedName{Namespace: vzNamespace, Name: vzName}, timeout, logFormat)
```
Comment (michael-cico) [inline tools/vz/cmd/install/install.go:217]: Do we need to update the `uninstall` command code also?

Comment (stevshap) [reply tools/vz/cmd/install/install.go:217]: +1

Comment (mgianatagh) [reply tools/vz/cmd/install/install.go:217]: I can address uninstall in a follow-up PR

File: tools/vz/cmd/helpers/vpo.go (hunk 2 above)
Comment (mgianatagh) [reply tools/vz/cmd/helpers/vpo.go:218]: how long do you feel we should wait?  1 or 2 minutes?

**Review by michael-cico — APPROVED** [review]

**Pull request merged**